### DISCUSSION TITLE
Moving to `serversideup/php:8.1-fpm-nginx` increased bootstrap times to 180ms

### DISCUSSION THREAD

**Discussion by dejagersh** [serversideup/docker-php#76] [Q&A]
Before I ran my Laravel app on `alpine` container on Fly.io.

Recently I moved to `serversideup/php:8.1-fpm-nginx` as it's much less code I have to manage. I am using the default Fly.io configuration that Fly provides for Laravel apps. However, when I run my app on `serversideup/php:8.1-fpm-nginx`, the bootstrapping times go up by A LOT. These are the p0.9 values:

<img width="939" alt="image" src="https://user-images.githubusercontent.com/9981548/192058346-d7f2342a-abbd-466e-9268-79326bc6a8d8.png">

and averages:

<img width="1227" alt="image" src="https://user-images.githubusercontent.com/9981548/192059026-ce6e0076-e24d-4088-b835-b722d5a47a22.png">

I deployed the container shortly after 19:00. This is what the trace is looking like:

<img width="901" alt="image" src="https://user-images.githubusercontent.com/9981548/192058709-bd0f652c-aad6-4d6f-bb41-0e8b6c863080.png">

As you can see, bootstrapping takes forever.

Anyone here have any idea why this could be? Or how I could debug it?

**Comment by jaydrogers** [comment]
Thanks for the report!

Out of curiosity, could you run these same tests with my beta image?

`serversideup/php:beta-8.1-fpm-nginx`

It’s a pretty major overhaul, but I have everything documented here: https://github.com/serversideup/docker-php/pull/59

**Reply by dejagersh** [reply]
Will do when I have time!

**Reply by jaydrogers** [reply]
Awesome! One other thing that came to mind too is my default is set to "ondemand".

I wonder if messing with these settings might help you too? I am interested in learning more on what you find: https://github.com/serversideup/docker-php/blob/dev/dist/8.1/fpm/etc/php/fpm/pool.d/y-override-php-defaults.conf#L69

**Reply by dejagersh** [reply]
I'll be sharing my experiences here :)

I just created a new Fly.io app using `serversideup/php:8.1-fpm-nginx`. Works fine.  After that I _only_ change `serversideup/php:8.1-fpm-nginx` => `serversideup/php:beta-8.1-fpm-nginx` and I get the following error:

```
2022-09-24T17:02:46.078 app[31c25651] ams [info] Starting init (commit: 249766e)...
2022-09-24T17:02:46.113 app[31c25651] ams [info] Preparing to run: `/entrypoint` as root
2022-09-24T17:02:46.156 app[31c25651] ams [info] 2022/09/24 17:02:46 listening on [fdaa:0:84cd:a7b:c207:31c2:5651:2]:22 (DNS: [fdaa::3]:53)
2022-09-24T17:02:47.911 app[31c25651] ams [info] s6-overlay-suexec: fatal: can only run as pid 1
2022-09-24T17:02:48.145 app[31c25651] ams [info] Starting clean up.
```

**Reply by dejagersh** [reply]
Trying `pm = dynamic` (on `serversideup/php:8.1-fpm-nginx`). I looked at my old configuration for `alpine` and I was 
using `pm = dynamic`.

Edit: didn't help anything.

<img width="339" alt="image" src="https://user-images.githubusercontent.com/9981548/192110824-a2517ca3-11da-427e-9134-081b41c80821.png">

**Reply by dejagersh** [reply]
The odd thing is that I can't reproduce this in a fresh Laravel project. There must be something in my project that doesn't work well on `serversideup/php:8.1-fpm-nginx` but works _fine_ on `alpine`. The app is so simple though...

**Reply by dejagersh** [reply]
But looking at some of my other projects: they get similarly terrible response times... It's at least 130ms for pages that don't even do DB queries

**Reply by jaydrogers** [reply]
Is it possibly related to this `openbase_dir` stuff? https://community.serversideup.net/t/bad-performance-of-laravel-based-wintercms/684/8?u=jaydrogers

**Reply by jaydrogers** [reply]
@thejager: Just following up on this because I am interested in figuring out what the problem is. Let me know if it is related to that discussion above with `openbase_dir` 👍
